### PR TITLE
add user,noauto to /opt/data mount

### PR DESCRIPTION
**Pull request by snopoke**
@kaapstorm

### CODE REVIEW

**Comment by snopoke** [comment]
@dannyroberts @TylerSheffels this was an issue on SoftLayer. The 'user,noauto' were missing from /etc/fstab which prevented the machines from coming up after boot (halted while trying to mount /opt/data)

File: ansible/roles/ecryptfs/tasks/main.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
 - name: Register mount configuration
   sudo: yes
-  shell: "tail /etc/mtab -n1"
+  shell: "cat /etc/mtab | grep '/opt/data' | sed -e 's/ rw/ user,noauto,rw/'"
```
Comment (snopoke) [inline ansible/roles/ecryptfs/tasks/main.yml]: Not that comfortable with this. Any better ideas?

Comment (benrudolph) [inline ansible/roles/ecryptfs/tasks/main.yml]: could you use [lineinfile](http://docs.ansible.com/ansible/lineinfile_module.html)? it looks like you're trying to change which users have access to `/opt/data` in that file, which seems like you could use that module

Comment (benrudolph) [inline ansible/roles/ecryptfs/tasks/main.yml]: what's supposed to go into `mount_configuration`?

Comment (snopoke) [inline ansible/roles/ecryptfs/tasks/main.yml]: suppose could have another task:
```
- name: add user, noauto
  when: "'noauto' not in mount_configuration.stdout"
  shell: "echo {{ mount_configuration.stdout }} | sed -e 's/ rw/ user,noauto,rw/'"
  register: mount_configuration
```

Comment (snopoke) [inline ansible/roles/ecryptfs/tasks/main.yml]: The issue is that the line we're adding is coming from the output of `tail /etc/mtab` but we want to alter the line before writing it to the file.

Comment (benrudolph) [inline ansible/roles/ecryptfs/tasks/main.yml]: oh i see. that is tricky. seems like `sed` is the easiest solution. it also seems odd though that it isn't there already when it adds `user,noauto` when you mount: https://github.com/dimagi/commcarehq-ansible/blob/ecryptfs/ansible/roles/ecryptfs/tasks/main.yml#L22

Comment (snopoke) [inline ansible/roles/ecryptfs/tasks/main.yml]: yea, it is a bit odd but its definitely not there

**Comment by dannyroberts** [comment]
Looks fine to me; @benrudolph might have more do say since he made the original setup

**Comment by kaapstorm** [comment]
@snopoke subsequently discovered that in Ubuntu 12.04 /etc/mtab includes the "user,noauto" options but Ubuntu 14.04 drops them for some reason.

**Comment by benrudolph** [comment]
it's @TylerSheffels's baby!

**Pull request merged**